content/courses/advanced-ada/chapters/types.rst

```diff
@@ -1643,19 +1643,913 @@ want to declare objects of this type.
+
+       Put_Line ("AF_1: "
+                 & AF_1'Image);
+
+       AF_2 := Float_Array_2 (AF_1);
+
+       Put_Line ("AF_2: "
+                 & AF_2'Image);
+    end Show_Conversion;
+
+As we can see in this example, the fact that :ada:`Integer_Array_1` and
+:ada:`Integer_Array_2` have the same component type (:ada:`Integer`) allows us
+to convert between them. The same applies to the :ada:`Float_Array_1` and
+:ada:`Float_Array_2` types.
+
+A conversion is not possible when the component types don't match. Even though
+we can convert between integer and floating-point types, we cannot convert an
+array of integers to an array of floating-point directly. Therefore, we cannot
+write a statement such as :ada:`AF_1 := Float_Array_1 (AI_1);`.
+
+However, when the components don't match, we can of course implement the array
+conversion by converting the individual components. For the example above, we
+used an iterated component association in an array aggregate:
+:ada:`[for I in AF_1'Range => Float (AI_1 (I))];`. (We discuss this topic later
+:ref:`in another chapter <Adv_Ada_Array_Aggregates>`.)
+
+We may also encounter array types originating from the instantiation of generic
+packages. In this case as well, we can use array conversions. Consider the
+following generic package:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Types.Generic_Array_Type_Conversion
+
+    generic
+       type T is private;
+    package Custom_Arrays is
+       type T_Array is array (Positive range <>) of T;
+    end Custom_Arrays;
+
+We could instantiate this generic package and reuse parts of the previous code
+example:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Types.Generic_Array_Type_Conversion
+
+    pragma Ada_2022;
+
+    with Ada.Text_IO;   use Ada.Text_IO;
+    with Custom_Arrays;
+
+    procedure Show_Conversion is
+       package CA_Int_1 is
+         new Custom_Arrays (T => Integer);
+       package CA_Int_2 is
+         new Custom_Arrays (T => Integer);
+
+       subtype Common_Range is Positive range 1 .. 3;
+
+       AI_1 : CA_Int_1.T_Array (Common_Range);
+       AI_2 : CA_Int_2.T_Array (Common_Range);
+    begin
+       AI_2 := [1, 2, 3];
+       AI_1 := CA_Int_1.T_Array (AI_2);
+
+       Put_Line ("AI_1: "
+                 & AI_1'Image);
+
+       AI_1 := [4, 5, 6];
+       AI_2 := CA_Int_2.T_Array (AI_1);
+
+       Put_Line ("AI_2: "
+                 & AI_2'Image);
+    end Show_Conversion;
+
+As we can see in this example, each of the instantiated :ada:`CA_Int_1` and
+:ada:`CA_Int_2` packages has a :ada:`T_Array` type. Even though these
+:ada:`T_Array` types have the same name, they're actually completely unrelated
+types. However, we can still convert between them in the same way as we did in
+the previous code examples.
+
+
+View conversion
+~~~~~~~~~~~~~~~
+
+As mentioned before, view conversions just allow names to be converted. Thus,
+we cannot use expressions in this case.
+
```

Comment:
I think it is important to relate the concept of "view" and then, in that context, say that the conversion changes the view of a given object to the target type's view, both in terms of components that exist and operations that are available. It doesn't really change anything at all in the value itself.

Reply:
Yes, I agree. (I hope the explanation that I've just added is clear enough.)